lib/middleware.js

```diff
@@ -0,0 +1,67 @@
+/* jshint node: true */
+'use strict';
+
+var Rule = require('./rule.js');
+
+module.exports = Middleware;
```

Comment:
It's also [the name for the concept in Hubot](https://hubot.github.com/docs/scripting/#middleware).